File: javascript/src/porta/PoliticaDisponibilidade.js
import { naoImplementado } from './_contrato.js';

// Regra de horario isolada: muda a politica sem tocar no servico de agendamento.
export class PoliticaDisponibilidade {
  estaLivre(medico, dataHora) {
    naoImplementado(this.constructor.name, 'estaLivre');
  }
}

File: java/src/br/unifor/facilitador/porta/PoliticaDisponibilidade.java
package br.unifor.facilitador.porta;

import br.unifor.facilitador.dominio.Medico;
import java.time.LocalDateTime;

/** Regra de horario isolada: muda a politica sem tocar no servico de agendamento. */
public interface PoliticaDisponibilidade {
    boolean estaLivre(Medico m, LocalDateTime dt);
}
